.github/workflows/simulation_test.yml

```diff
@@ -237,6 +237,10 @@ jobs:
     - name: Install test dependencies
       run: |
         pip install -r tests/requirements.txt
+        pip install flake8 flake8-pyproject
+
+    - name: Run flake8
+      run: flake8 .
```

Comment:
Can you run flake independently on the sources only with no installation? As in separate job with separate status?

Also, `ruff` may be a better all-in-one linter.